test/assocWith.js

```diff
@@ -0,0 +1,19 @@
+const { expect } = require('chai')
+const property   = require('prop-factory')
+const always     = require('ramda/src/always')
+
+const { assocWith } = require('..')
+
+const assigner = assocWith('foo', always('bar'))
```

Comment:
Your tests are only passing because you have a constant function here.  Instead of `always('bar')`, create a new function that takes the original object and returns a value.